quixstreams/sources/base.py

```diff
@@ -0,0 +1,285 @@
+                        value="foo",
+                    )
+                    time.sleep(1)
+
+            def stop(self):
+                self._running = False
+        ```
+        """
+
+    @abstractmethod
+    def default_topic(self) -> Topic:
+        """
+        This method is triggered when the user hasn't specified a topic for the source.
+
+        The source should return a default topic configuration
+        """
+
+
+class Source(BaseSource):
+    """
+    BaseSource class implementation providing
+
+    Implementation for the abstract method:
+        * `default_topic`
+
+    Helper methods
+        * serialize
+        * produce
+        * flush
+    """
+
+    def __init__(self, name: str, shutdown_timeout: float = 10) -> None:
+        """
+        :param name: The source unique name. Used to generate the topic configurtion
+        :param shutdown_timeout: Time in second the application waits for the source to gracefully shutdown
+        """
+        super().__init__()
+
+        # used to generate a unique topic for the source.
+        self.name = name
+
+        self.shutdown_timeout = shutdown_timeout
+
+    def serialize(
+        self,
+        key: Optional[object] = None,
+        value: Optional[object] = None,
+        headers: Optional[Headers] = None,
+        timestamp_ms: Optional[int] = None,
+    ) -> KafkaMessage:
+        """
+        Serialize data into a :class:`quixstreams.models.messages.KafkaMessage` using the producer topic serializers.
+
+        :return: :class:`quixstreams.models.messages.KafkaMessage`
+        """
+        return self._producer_topic.serialize(
+            key=key, value=value, headers=headers, timestamp_ms=timestamp_ms
+        )
+
+    def produce(
+        self,
+        value: Optional[Union[str, bytes]] = None,
+        key: Optional[Union[str, bytes]] = None,
+        headers: Optional[Headers] = None,
+        partition: Optional[int] = None,
+        timestamp: Optional[int] = None,
+        poll_timeout: float = 5.0,
+        buffer_error_max_tries: int = 3,
+    ) -> None:
+        """
+        Produce data to kafka using the source topic.
+        """
+
+        self._producer.produce(
+            topic=self._producer_topic.name,
+            value=value,
+            key=key,
+            headers=headers,
+            partition=partition,
+            timestamp=timestamp,
+            poll_timeout=poll_timeout,
+            buffer_error_max_tries=buffer_error_max_tries,
+        )
+
+    def flush(self, timeout: Optional[float] = None) -> None:
+        """
+        This method flush the producer.
+
+        It ensure all message are successfully delived to kafka
+
+        :param float timeout: time to attempt flushing (seconds).
+            None use producer default or -1 is infinite. Default: None
+
+        :raises CheckpointProducerTimeout: if any message fails to produce before the timeout
+        """
+        logger.debug("Flushing source")
```

Comment:
Let's add the source name here so it's easier to identify it in logs

Reply:
THe logger is already configured to display the source name